aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/StorageComponentTest.java

```diff
@@ -0,0 +1,235 @@
+     */
+    @Test
+    public void testDownloadToFileGetsFile() throws StorageException {
+        final String fromRemoteKey = RandomString.string();
+        final String toLocalPath = RandomString.string();
+
+        // Since we use a mock StorageService, it will return a null
+        // result by default. We need a non-null transfer observer.
+        // One option is to mock that, too.
+        TransferObserver observer = mock(TransferObserver.class);
+        when(storageService.downloadToFile(anyString(), any(File.class)))
+                .thenReturn(observer);
+
+        // Since we use a mock TransferObserver, it has no internal logic
+        // to know to call back the listener! So, we simulate the success
+        // callback, as part of our "happy path" test.
+        doAnswer(invocation -> {
+            TransferListener listener = invocation.getArgument(0);
+            listener.onStateChanged(0, TransferState.COMPLETED);
+            return null;
+        })
+                .when(observer)
+                .setTransferListener(any(TransferListener.class));
+
+        StorageDownloadFileResult result =
+                Await.<StorageDownloadFileResult, StorageException>result((onResult, onError) ->
+                    storage.downloadFile(
+                        fromRemoteKey,
+                        toLocalPath,
+                        onResult,
+                        onError
+                    )
+                );
+
+        assertEquals(toLocalPath, result.getFile().toString());
+    }
+
+    /**
+     * Test that calling upload file method from Storage category correctly
+     * invokes the registered AWSS3StoragePlugin instance and returns a
+     * {@link StorageUploadFileResult} with correct remote key.
+     * @throws StorageException when an error is encountered while uploading
+     */
+    @Test
+    public void testUploadFileGetsKey() throws StorageException {
+        final String toRemoteKey = RandomString.string();
+        final String fromLocalPath = RandomString.string();
+
+        // Since we use a mock StorageService, it will return a null
+        // result by default. We need a non-null transfer observer.
+        // One option is to mock that, too.
+        TransferObserver observer = mock(TransferObserver.class);
+        when(storageService.uploadFile(anyString(), any(File.class)))
+                .thenReturn(observer);
+
+        // Since we use a mock TransferObserver, it has no internal logic
+        // to know to call back the listener! So, we simulate the success
+        // callback, as part of our "happy path" test.
+        doAnswer(invocation -> {
+            TransferListener listener = invocation.getArgument(0);
+            listener.onStateChanged(0, TransferState.COMPLETED);
+            return null;
+        })
+                .when(observer)
+                .setTransferListener(any(TransferListener.class));
```

Comment:
Ugh, yea. Does checkstyle not like it as:
```
    })
    .alignedToSameCol()
```
?
I think I have been getting around this by dropping the first argument to a newline and adding +4 ...
```
doAnswer(
    invocation -> {
    })
    .when(observer)
    .setTransfer(...)
```